### PR TITLE
Added email confirmation, password reset and change password endpoints

### PR DESCRIPTION
**Pull request by Jaffenheimer**
The application now sends an email for verification (not done, need front-end implementation) sends an email if you request you password to be reset, and created support for changing of password as well.

### CODE REVIEW

File: P7Internet.Persistence/UserRepository/UserRepository.cs [hunk 1]
```diff
@@ -33,55 +33,110 @@ private async Task<User> GetUser(string name)
...
+        var salt = GenerateSalt();
+        var passwordHash = GenerateHash(password + salt);
+        var result = await Connection.ExecuteAsync(query,
+            new {Name = userName, Password_hash = passwordHash, Password_salt = salt, Updated = DateTime.UtcNow});
+        return result > 0;
+    }
+
+    public async Task<bool> ChangePassword(string userName, string oldPassword, string newPassword)
+    {
+        var checkIfUserExist = await GetUser(userName);
```
Comment (JJTM20) [inline P7Internet.Persistence/UserRepository/UserRepository.cs]: Why not just "user"?

File: P7Internet.Persistence/UserRepository/UserRepository.cs [hunk 2]
```diff
@@ -33,55 +33,110 @@ private async Task<User> GetUser(string name)
...
+    public async Task<bool> ConfirmEmail(string userName, string emailAddress)
+    {
+        var query = $@"UPDATE {TableName} SET EmailConfirmed = true WHERE Name = @userName AND Email = @email";
+        var result = await Connection.ExecuteAsync(query, new {Name = userName, Email = emailAddress});
+        return result > 0;
+    }
+
+    public async Task<bool> ResetPassword(string userName, string password)
+    {
+        var checkIfUserExist = await GetUser(userName);
```
Comment (JJTM20) [inline P7Internet.Persistence/UserRepository/UserRepository.cs]: Same as below

File: P7Internet.Persistence/UserRepository/UserRepository.cs [hunk 3]
```diff
@@ -33,55 +33,110 @@ private async Task<User> GetUser(string name)
...
             }
         }
+
         return null;
     }
 
+    public async Task<bool> ConfirmEmail(string userName, string emailAddress)
+    {
+        var query = $@"UPDATE {TableName} SET EmailConfirmed = true WHERE Name = @userName AND Email = @email";
+        var result = await Connection.ExecuteAsync(query, new {Name = userName, Email = emailAddress});
```
Comment (JJTM20) [inline P7Internet.Persistence/UserRepository/UserRepository.cs:85]: "Name" should be removed and and "Email" should be "email" to correspond to the query parameters.

**Review by JJTM20 — CHANGES_REQUESTED** [review]

File: P7Internet.Persistence/UserRepository/UserRepository.cs (hunk 3 above)
Comment (Jaffenheimer) [reply P7Internet.Persistence/UserRepository/UserRepository.cs:85]: Ændret

File: P7Internet.Persistence/UserRepository/UserRepository.cs (hunk 2 above)
Comment (Jaffenheimer) [reply P7Internet.Persistence/UserRepository/UserRepository.cs]: Ændret

File: P7Internet.Persistence/UserRepository/UserRepository.cs (hunk 1 above)
Comment (Jaffenheimer) [reply P7Internet.Persistence/UserRepository/UserRepository.cs]: ændret

**Review by JJTM20 — APPROVED** [review]

**Pull request merged**